util/manual_control_connect.bash

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash 
+
+display_help()
+{
+    echo "Usage: $0 [-h] [-r 0] [-c keyboard-controller] [-C True]"
+    echo 
+    echo "Options:"
+    echo "  -h   Show this help message."
+    echo "  -r   Control robot_id manually (default: 0)."
+    echo "  -c   Connect controller (default: keyboard-controller)."
+    echo "  -C   Connect flag, set to False to disconnect (default: True)."
+    echo
+    echo "(Note: at least 1 arg must be given to avoid showing this help message.)"
+    # never do anything after displaying help msg
+    exit 0
+}
+
+# if no args given, display help msg above
+if [[ $# -eq 0 ]] ; then
+  display_help
+fi
+
+# if -h is first flag, display help msg above and exit
+# otherwise, take in args per the help msg above, then call the ros2 service
+robot_id=0
+controller_uuid="keyboard-controller"
+connect=True
+
+while getopts h:r:c:C: flag
+do
+    case "${flag}" in
+        h) display_help;;
+        r) robot_id=${OPTARG};;
+        c) controller_uuid=${OPTARG};;
+        C) connect=${OPTARG};;
+    esac
+done
+
+cmd="ros2 service call /select_manual rj_msgs/srv/SetManual \"{robot_id: $robot_id, controller_uuid: $controller_uuid, connect: $connect}\""
```

Comment:
This is a big improvement from before, but I would add a card to have the ui call this script.

Reply:
(When you input a manual id and launch in manual mode)

Reply:
way ahead of you: https://app.clickup.com/t/2tdhtqv